server/src/main/java/de/pnp/manager/component/upgrade/UpgradeRecipe.java

```diff
@@ -0,0 +1,92 @@
+package de.pnp.manager.component.upgrade;
+
+import de.pnp.manager.component.DatabaseObject;
+import de.pnp.manager.component.IRecipeEntry;
+import java.util.Collection;
+import java.util.Objects;
+import org.bson.types.ObjectId;
+import org.springframework.data.mongodb.core.mapping.DBRef;
+
+/**
+ * A recipe to create an {@link Upgrade}.
+ */
+public class UpgradeRecipe extends DatabaseObject {
+
+    /**
+     * The product of this {@link UpgradeRecipe}.
+     */
+    @DBRef
+    private final Upgrade upgrade;
+
+    /**
+     * The {@link Upgrade upgrads} which already need to be present on the item to use this {@link UpgradeRecipe}.
+     */
+    @DBRef
+    private final Collection<Upgrade> necessaryUpgrades;
+
+    /**
+     * Additional requirements needed to use this {@link UpgradeRecipe}.
+     */
+    private final String requirement;
+
+    /**
+     * Other circumstances needed to use this {@link UpgradeRecipe}.
+     */
+    private final String otherCircumstances;
```

Comment:
create inner class UpgradeRecipe.UpgradeRequirements with the three things

Reply:
Why? I dont see a scenario where we reuse that part.